### PR TITLE
Adding section about Truffle Develop

### PR DESCRIPTION
**Pull request by bmmpxf**
This PR alters the documentation page on the console to also include details about Truffle Develop.

### CODE REVIEW

File: public/docs/getting_started/console.md [hunk 1]
```diff
@@ -1,36 +1,137 @@
-# Using the console
+# Using Truffle Develop and the console
 
-Sometimes it's nice to work with your contracts interactively for testing and debugging purposes, or for executing transactions by hand. Truffle provides you an easy way to do this via an interactive console, with your contracts available and ready to use.
+Sometimes it's nice to work with your contracts interactively for testing and debugging purposes, or for executing transactions by hand. Truffle provides you two easy ways to do this via an interactive console, with your contracts available and ready to use.
 
-## Command
+* **Truffle console**: A basic interactive console
```
Comment (tcoulter) [inline public/docs/getting_started/console.md]: Probably should add 
> connecting to any Ethereum client

File: public/docs/getting_started/console.md [hunk 2]
```diff
@@ -1,36 +1,137 @@
...
+Reasons to use Truffle console:
+
+* You have a client you're already using, such as [Ganache](/docs/ganache/using) or geth
+* You want to migrate to a testnet (or the main Ethereum network)
+* You want to use a specific mnemonic or account list
+
+Reasons to use Truffle Develop:
+
+* You are testing your project with no intention of immediately deploying
+* You don't need to work with specific accounts
```
Comment (tcoulter) [inline public/docs/getting_started/console.md]: Maybe in parentheses here, say: 
> (and you're happy using default development accounts)

File: public/docs/getting_started/console.md [hunk 3]
```diff
@@ -1,36 +1,137 @@
...
 
-Sometimes it's nice to work with your contracts interactively for testing and debugging purposes, or for executing transactions by hand. Truffle provides you an easy way to do this via an interactive console, with your contracts available and ready to use.
+Sometimes it's nice to work with your contracts interactively for testing and debugging purposes, or for executing transactions by hand. Truffle provides you two easy ways to do this via an interactive console, with your contracts available and ready to use.
 
-## Command
+* **Truffle console**: A basic interactive console
+* **Truffle Develop**: An interactive console that also spawns a development blockchain
 
-To fire up the console, simply run:
+## Why two different consoles?
```
Comment (tcoulter) [inline public/docs/getting_started/console.md]: Love this section.

File: public/docs/getting_started/console.md [hunk 4]
```diff
@@ -1,36 +1,137 @@
...
+
+* You have a client you're already using, such as [Ganache](/docs/ganache/using) or geth
+* You want to migrate to a testnet (or the main Ethereum network)
+* You want to use a specific mnemonic or account list
+
+Reasons to use Truffle Develop:
+
+* You are testing your project with no intention of immediately deploying
+* You don't need to work with specific accounts
+* You don't need or want a separate blockchain running
```
Comment (tcoulter) [inline public/docs/getting_started/console.md]: You still have a separate blockchain when using `truffle develop`, but you just don't have to manage it. I'd change it to: 

> You don't want to install and manage a separate blockchain client

Or something like that.

File: public/docs/getting_started/console.md [hunk 5]
```diff
@@ -1,36 +1,137 @@
...
 
-This will load up a console using the `development` network, if specified. You can override this using the `--network` option. See more details in the [Networks](/docs/advanced/networks) section as well as the [command reference](/docs/advanced/commands).
+## Commands
 
-When you load the console, you'll immediately see output like this:
+All commands require that you be in the root of your project folder.
```
Comment (tcoulter) [inline public/docs/getting_started/console.md]: I don't think this is actually true. Truffle is smart enough to look up into your directories for a `truffle.js` file. Give it a shot and let me know what you find. e.g., if you `cd` into your `contracts` directory, then run `truffle console`, it _should_ load.

Comment (bmmpxf) [reply public/docs/getting_started/console.md]: Ah right you are. I ran in the `contracts` directory and it ran just fine. Good to know.

**Comment by bmmpxf** [comment]
Awesome! Thanks for the review. :)

**Pull request merged**